### PR TITLE
Feature/instanceOf

### LINKED ISSUE #125: Add `instanceOf` rule
This rule should reflect the usage of the `instanceof` operator:

```javascript
const date = new Date();

v8n().instanceOf(Date).test(date); // true
```

### PR DESCRIPTION
**Pull request by tomraithel**
## Description

Add `instanceOf` rule.

Fixes #125

In addition i realized, that there was no test for the `object()` rule. I added some test-cases so that the tests still pass but I´m not sure if it´s intented, that f.e. `[1,3,4]` should pass the the `object()` rule. If that is not the case, I would suggest opening a new bug.

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] Other: Add missing test for `object` validator

## Checklist:

- [x] I have created my branch from a recent version of `master`
- [x] The code is clean and easy to understand
- [x] I have made corresponding changes to the documentation
- [x] I have added changes to the `Unreleased` section of the CHANGELOG

### CODE REVIEW

File: docs/api/README.md [hunk 1]
```diff
@@ -479,6 +479,24 @@ sidebar: auto
     .test({ key: "value" }); // true
   ```
 
+### instanceOf
+
+- **Signature:** `instanceOf()`
+
+- **Usage:**
+
+  This rule verifies that the prototype of the tested value appears anywhere in the prototype chain of the provided costructor.
```
Comment (imbrn) [inline docs/api/README.md]: Just a little typo here: _costructor_ -> **constructor**

File: src/v8n.test.js [hunk 2]
```diff
@@ -464,6 +464,32 @@ describe("rules", () => {
...
+  test("instanceOf", () => {
+    const validationDate = v8n().instanceOf(Date);
+    expect(validationDate.test(new Date())).toBeTruthy();
+    expect(validationDate.test({})).toBeFalsy();
+    expect(validationDate.test(null)).toBeFalsy();
+    expect(validationDate.test("string")).toBeFalsy();
+
+    const validationObject = v8n().instanceOf(Object);
+    expect(validationObject.test(new Date())).toBeTruthy();
+    expect(validationObject.test(new Date())).toBeTruthy();
```
Comment (imbrn) [inline src/v8n.test.js]: Duplicated line here

**Comment by tomraithel** [comment]
@imbrn I fixed the typos

**Comment by imbrn** [comment]
Thank you @tomraithel. That's great!

**Pull request merged**